pkg/controller/deployment/deployment_controller.go

```diff
@@ -83,7 +83,41 @@ func (d *DeploymentController) reconcileDeployment(deployment *extensions.Deploy
 }
 
 func (d *DeploymentController) reconcileRecreateDeployment(deployment extensions.Deployment) error {
-	// TODO: implement me.
+	newRC, err := d.getNewRC(deployment)
+	if err != nil {
+		return err
+	}
+
+	oldRCs, err := d.getOldRCs(deployment)
+	if err != nil {
+		return err
+	}
+
+	allRCs := append(oldRCs, newRC)
+
+	// scale down old rcs
+	scaledDown, err := d.scaleDownOldRCs(oldRCs, deployment)
+	if err != nil {
+		return err
+	}
+	if scaledDown {
+		// Update DeploymentStatus
+		err := d.updateDeploymentStatus(allRCs, newRC, deployment)
+		if err != nil {
+			return err
+		}
+	}
+
+	// scale up new rc
+	scaledUp, err := d.scaleUpNewRC(newRC, deployment)
+	if err != nil {
+		return err
+	}
+	if scaledUp {
+		// Update DeploymentStatus
+		return d.updateDeploymentStatus(allRCs, newRC, deployment)
```

Comment:
Should we just return after `scaleDown` and `updateDeploymentStatus`, leaving `scaleUp` in the next sync time, just like `reconcileRollingUpdateDeployment` do or update `oldRCs` and `newRC` before we call `updateDeploymentStatus`